File: exercises/day10-TimePortal/csharp/TimePortal/TimePortal/PortalNetwork.cs
namespace TimePortal;

public class PortalNetwork
{
    private List<TimePortal> _portals;
    private string _networkName;

    public PortalNetwork(string networkName)
    {
        _networkName = networkName;
        _portals = new List<TimePortal>();
    }

    public void AddPortal(TimePortal portal)
    {
        _portals.Add(portal);
    }

    public int GetTotalEnergy()
    {
        var total = 0;
        for (var i = 0; i < _portals.Count; i++)
        {
            total = total + _portals[i].EnergyLevel;
        }
        return total;
    }

    public TimePortal? FindActivePortalWithEnergy(int minimumEnergy)
    {
        for (var i = 0; i < _portals.Count; i++)
        {
            if (_portals[i].IsActive && _portals[i].EnergyLevel >= minimumEnergy)
            {
                return _portals[i];
            }
        }
        return null;
    }

    public string NetworkName => _networkName;
    public List<TimePortal> Portals => _portals;
}

File: exercises/day10-TimePortal/typescript/src/PortalNetwork.ts
import { TimePortal } from './TimePortal';

export class PortalNetwork {
    private _networkName: string;
    private _portals: TimePortal[];

    constructor(networkName: string) {
        this._networkName = networkName;
        this._portals = [];
    }

    addPortal(portal: TimePortal): void {
        this._portals.push(portal);
    }

    getTotalEnergy(): number {
        let total = 0;
        for (let i = 0; i < this._portals.length; i++) {
            total = total + this._portals[i].energyLevel;
        }
        return total;
    }

    findActivePortalWithEnergy(minimumEnergy: number): TimePortal | null {
        for (let i = 0; i < this._portals.length; i++) {
            if (this._portals[i].isActive && this._portals[i].energyLevel >= minimumEnergy) {
                return this._portals[i];
            }
        }
        return null;
    }

    get networkName(): string {
        return this._networkName;
    }

    get portals(): TimePortal[] {
        return this._portals;
    }
}
